keyboards/moonlander/moonlander.c

```diff
@@ -276,39 +276,21 @@ led_config_t g_led_config = { {
     {  NO_LED, 64, 59, 54, 49, 44, 39 },
     {  NO_LED, NO_LED, 60, 55, 50, 45, 40 },
     {  NO_LED, NO_LED, NO_LED, 71, 70, 69, 68 },
-}, {
-    {   0,   0 }, {   0,  12 }, {   0,  25 }, {   0,  38 }, {   0,  51 },
-    {  17,   0 }, {  17,  12 }, {  17,  25 }, {  17,  38 }, {  17,  51 },
-    {  34,   0 }, {  34,  12 }, {  34,  25 }, {  34,  38 }, {  34,  51 },
-    {  51,   0 }, {  51,  12 }, {  51,  25 }, {  51,  38 }, {  51,  51 },
-    {  68,   0 }, {  68,  12 }, {  68,  25 }, {  68,  38 }, {  68,  51 },
-    {  86,   0 }, {  86,  12 }, {  86,  25 }, {  86,  38 },
-    { 105,   0 }, { 105,  12 }, { 105,  25 },
-    {  90,  55 }, { 105,  68 }, { 116,  86 }, { 116, 59 },
-
-    { 250,   0 }, { 250,  12 }, { 250,  25 }, { 250,  38 }, { 250,  51 },
-    { 233,   0 }, { 233,  12 }, { 233,  25 }, { 233,  38 }, { 233,  51 },
-    { 216,   0 }, { 216,  12 }, { 216,  25 }, { 216,  38 }, { 216,  51 },
-    { 198,   0 }, { 198,  12 }, { 198,  25 }, { 198,  38 }, { 198,  51 },
-    { 181,   0 }, { 181,  12 }, { 181,  25 }, { 181,  38 }, { 181,  51 },
-    { 163,   0 }, { 163,  12 }, { 163,  25 }, { 163,  38 },
-    { 146,   0 }, { 146,  12 }, { 146,  25 },
-    { 161,  55 }, { 161,  68 }, { 146,  86 }, { 131, 59 }
-
-}, {
-    1, 1, 1, 1, 1, 4,
-    4, 4, 4, 1, 4, 4,
-    4, 4, 1, 4, 4, 4,
-    4, 1, 4, 4, 4, 4,
-    1, 4, 4, 4, 4, 4,
-    4, 4, 1, 1, 1, 1,
-    1, 1, 1, 1, 1, 4,
-    4, 4, 4, 1, 4, 4,
-    4, 4, 1, 4, 4, 4,
-    4, 1, 4, 4, 4, 4,
-    1, 4, 4, 4, 4, 4,
-    4, 4, 1, 1, 1, 1
-} };
+}, LED_LAYOUT_moonlander(
```

Comment:
Unfortunately, `qmk lint` (`lib/python/qmk/c_parse.py`) does not like the use of macros here. Reverting would be easy, but using this macro brought to light some issues, so there is value in using a macro here.

Reply:
See this thread https://github.com/qmk/qmk_firmware/pull/18973#discussion_r1018571012

Reply:
I don't believe `g_led_config` can be represented in `info.json` yet, can it? I guess the benefits provided by using a macro here (i.e. catching mapping oddities) are out of reach to the DD approach, or are there some cool plans in this direction?

Reply:
It's possible, but a lot of the data-driven stuff is not well documented just yet, and some things are only in develop (which should merge in a day or so).
https://github.com/qmk/qmk_firmware/blob/master/keyboards/1upkeyboards/pi60_hse/info.json#L31-L51
https://github.com/qmk/qmk_firmware/blob/master/data/schemas/keyboard.jsonschema#L294-L320

Reply:
Hmm, unfortunately the order in the json file is the LED indexing order. This precludes generating a layout macro from them automatically. Automatic generation on the basis of a key layout _is_ possible, but only if every led has a position on the matrix that also has a key assigned to it. Even then, it is quite involved and we don't have a way to configure the name of the macro (`LED_{layout_name}` is possible, though).

I guess for now this leaves no other option than to set the LED layout macro the old fashioned way in `moonlander.h`, but revert its usage in defining `g_led_config`. I'll do that once I know what to do with the weird positions currently set in `g_led_config`.